prov/efa/test/efa_unit_test_ep.c

```diff
@@ -138,10 +138,14 @@ void test_efa_rdm_ep_handshake_exchange_host_id(struct efa_resource **state, uin
 	pkt_attr.host_id = g_efa_unit_test_mocks.peer_host_id;
 	efa_unit_test_handshake_pkt_construct(pkt_entry, &pkt_attr);
 
-	/* this mock will save the send work request (wr) in a global linked list */
-	ibv_qp = efa_rdm_ep->base_ep.qp->ibv_qp;
-	ibv_qp->context->ops.post_send = &efa_mock_ibv_post_send_verify_handshake_pkt_local_host_id_and_save_wr;
-	expect_function_call(efa_mock_ibv_post_send_verify_handshake_pkt_local_host_id_and_save_wr);
+	ibv_qp = efa_rdm_ep->base_ep.qp->ibv_qp_ex;
+	ibv_qp->wr_start = &efa_mock_ibv_wr_start_no_op;
+	/* this mock will save the send work request (wr) in a global array */
+	ibv_qp->wr_send = &efa_mock_ibv_wr_send_verify_handshake_pkt_local_host_id_and_save_wr;
+	ibv_qp->wr_set_inline_data_list = &efa_mock_ibv_wr_set_inline_data_list_no_op;
+	ibv_qp->wr_set_ud_addr = &efa_mock_ibv_wr_set_ud_addr_no_op;
+	ibv_qp->wr_complete = &efa_mock_ibv_wr_complete_no_op;
```

Comment:
Suggest asserting `efa_mock_ibv_wr_*` were called

Reply:
I does not think it is a good idea. Some of the functions can be called in different order. `set_ud_addr` and `set_inline_data_list`.